### PR TITLE
zoltan: add optional dependency on parmetis

### PR DESCRIPTION
**Pull request by cwsmith**
add parmetis support to zoltan for use of multi-level graph based partitioning procedures

### CODE REVIEW

File: var/spack/repos/builtin/packages/zoltan/package.py [hunk 1]
```diff
@@ -53,9 +53,13 @@ class Zoltan(Package):
 
     variant('fortran', default=True, description='Enable Fortran support.')
     variant('mpi', default=True, description='Enable MPI support.')
+    variant('parmetis', default=False, description='Enable ParMETIS support.')
 
     depends_on('mpi', when='+mpi')
 
+    depends_on('parmetis@4:', when='+parmetis')
+    depends_on('mpi', when='^parmetis')
```
Comment (davydden) [inline var/spack/repos/builtin/packages/zoltan/package.py]: parmetis will bring `mpi` as its dependency, so there is no need to have this line here.

Comment (cwsmith) [reply var/spack/repos/builtin/packages/zoltan/package.py]: OK.  Thank you.

I have broken the long lines the Flake8 test found and will remove line 61.  I assume that I should force push these changes to my branch so they appear as one commit?

Comment (davydden) [reply var/spack/repos/builtin/packages/zoltan/package.py]: >  I should force push these changes to my branch so they appear as one commit?

yes, if you squashed already the changes with the original commit, you need to push with `-f`.

Comment (davydden) [reply var/spack/repos/builtin/packages/zoltan/package.py]: p.s. you probably should add `conflicts('+parmetis', when='~mpi')` as likely it does not make sense to build it with `~mpi+parmetis`.

Comment (cwsmith) [reply var/spack/repos/builtin/packages/zoltan/package.py]: Thanks again. I just force pushed to my fork.  Locally executing `spack test` produces `918 passed, 75 skipped in 5830.47 seconds`.

**Comment by cwsmith** [comment]
one more commit... my python line continuations were incorrect

**Pull request merged**